src/Lumi/Components2/Banner.purs

```diff
@@ -0,0 +1,155 @@
+module Lumi.Components2.Banner
+  ( Banner
+  , BannerProps
+  , banner
+  , actionBanner
+  , primary
+  , active
+  , warning
+  , error
+  , module Styles.Banner
+  ) where
+
+import Prelude
+
+import Color (lighten)
+import Data.Foldable (fold)
+import Data.Monoid as Monoid
+import Data.Tuple.Nested ((/\))
+import Effect.Unsafe (unsafePerformEffect)
+import Lumi.Components (LumiComponent, PropsModifier, lumiComponent, lumiElement)
+import Lumi.Components.Icon as Icon
+import Lumi.Components.Spacing (Space(..))
+import Lumi.Components2.Box (box)
+import Lumi.Styles (color, css, nested, prop, str, styleModifier, styleModifier_)
+import Lumi.Styles.Banner (banner) as S
+import Lumi.Styles.Border (_listSpaced, _listCompact) as Styles.Banner
+import Lumi.Styles.Box (FlexAlign(..), _align, _column, _fill, _interactive, _justify, _row) as S
+import Lumi.Styles.Responsive (desktopQuery)
+import Lumi.Styles.Responsive (onDesktop) as S
+import Lumi.Styles.Theme (LumiTheme(..), useTheme)
+import React.Basic (JSX)
+import React.Basic.DOM as R
+import React.Basic.DOM.Events (capture_)
+import React.Basic.Hooks (useState)
+import React.Basic.Hooks as React
+
+data Banner = Banner
+
+type BannerProps =
+  ( component :: Banner
```

Comment:
We don't export the single constructor of the `Banner` data type so a banner can be identified in a `PropsModifier` as `forall props. PropsModifier (component :: Banner | props)`.

Reply:
It is necessary if `actionBanner` is a `LumiComponent`, so that we can apply all of the modifiers that work on `banner` on it too.

Reply:
I'm not sure I follow.. wouldn't it work on any component with the same prop shape?

Reply:
Yes, but `actionBanner` doesn't have the same prop shape as `banner`:
```purescript
type BannerProps =
  ( dismissable :: Boolean
  , content :: Array JSX
  )

type ActionBannerProps =
  ( dismissable :: Boolean
  , actions :: Array JSX
  , content :: Array JSX
  )
```